dnf/artifact-registry.py

```diff
@@ -36,6 +36,11 @@ def config(self):
       # We don't have baseurl option so skip it earlier.
       if not hasattr(repo, 'baseurl'):
         continue
+      # Check if the custom 'ar_auth' option is set in the repository's config.
```

Comment:
This check could go before the check for "baseurl" (line 37); that's just a null-guard before accessing the baseurl property. Since your check doesn't care about baseurl, it should go before the check which is only relevant to later checks.

Reply:
Thanks for reviewing! I went with `artifact_registry_oauth`, the google prefix feels a little too verbose :)

Tested:

```shell
[rbarash_google_com@quickstart-yum-vm ~]$ sudo yum makecache
Loaded plugins: artifact-registry, fastestmirror
epel/x86_64/metalink                                                                                                                                               | 5.1 kB  00:00:00
Loading mirror speeds from cached hostfile
 * epel: d2lzkl7pfhq30w.cloudfront.net
base                                                                                                                                                               | 3.6 kB  00:00:00
extras                                                                                                                                                             | 2.9 kB  00:00:00
google-cloud-sdk                                                                                                                                                   | 1.4 kB  00:00:00
google-compute-engine                                                                                                                                              | 1.4 kB  00:00:00
updates                                                                                                                                                            | 2.9 kB  00:00:00
yum-quickstart                                                                                                                                                     | 1.4 kB  00:00:00
Metadata Cache Created

[rbarash_google_com@quickstart-yum-vm ~]$ cd artifact-registry-yum-plugin/

[rbarash_google_com@quickstart-yum-vm artifact-registry-yum-plugin]$ git pull
remote: Enumerating objects: 10, done.
remote: Counting objects: 100% (10/10), done.
remote: Compressing objects: 100% (1/1), done.
remote: Total 6 (delta 3), reused 6 (delta 3), pack-reused 0 (from 0)
Unpacking objects: 100% (6/6), done.
From https://github.com/rafibarash/artifact-registry-yum-plugin
   465808d..9886fb4  main       -> origin/main
Updating 465808d..9886fb4
Fast-forward
 dnf/artifact-registry.py | 8 ++++----
 yum/artifact-registry.py | 4 ++--
 2 files changed, 6 insertions(+), 6 deletions(-)

[rbarash_google_com@quickstart-yum-vm artifact-registry-yum-plugin]$ sudo cp yum/artifact-registry.py /usr/lib/yum-plugins/artifact-registry.py

[rbarash_google_com@quickstart-yum-vm artifact-registry-yum-plugin]$ sudo yum makecache
Loaded plugins: artifact-registry, fastestmirror
epel/x86_64/metalink                                                                                                                                               | 5.1 kB  00:00:00
Loading mirror speeds from cached hostfile
 * epel: d2lzkl7pfhq30w.cloudfront.net
base                                                                                                                                                               | 3.6 kB  00:00:00
extras                                                                                                                                                             | 2.9 kB  00:00:00
google-cloud-sdk                                                                                                                                                   | 1.4 kB  00:00:00
google-compute-engine                                                                                                                                              | 1.4 kB  00:00:00
updates                                                                                                                                                            | 2.9 kB  00:00:00
https://us-east7-yum.kavala-team.dev/prod/quickstart-yum-repo/repodata/repomd.xml: [Errno 14] HTTPS Error 403 - Forbidden
Trying other mirror.
To address this issue please refer to the below wiki article

https://wiki.centos.org/yum-errors

If above article doesn't help to resolve this issue please use https://bugs.centos.org/.

Metadata Cache Created

[rbarash_google_com@quickstart-yum-vm artifact-registry-yum-plugin]$ sudo vim /etc/yum.repos.d/artifact-registry.repo

[rbarash_google_com@quickstart-yum-vm artifact-registry-yum-plugin]$ sudo yum makecache
Loaded plugins: artifact-registry, fastestmirror
epel/x86_64/metalink                                                                                                                                               | 5.1 kB  00:00:00
Loading mirror speeds from cached hostfile
 * epel: d2lzkl7pfhq30w.cloudfront.net
base                                                                                                                                                               | 3.6 kB  00:00:00
extras                                                                                                                                                             | 2.9 kB  00:00:00
google-cloud-sdk                                                                                                                                                   | 1.4 kB  00:00:00
google-compute-engine                                                                                                                                              | 1.4 kB  00:00:00
updates                                                                                                                                                            | 2.9 kB  00:00:00
yum-quickstart                                                                                                                                                     | 1.4 kB  00:00:00
Metadata Cache Created
```